eta/core/types.py

```diff
@@ -676,3 +718,195 @@ class FrameSequence(JSONFileSequence):
     '''
 
     pass
+
+
+class Features(FileSequence, ConcreteData):
+    '''A sequence of features indexed by one numeric parameter.
+
+    Examples:
+        /path/to/features/%05d.npz
+    '''
+
+    @staticmethod
+    def gen_path(basedir, params):
+        return os.path.join(
+            basedir, "{name}", "{idx}.npz").format(**params)
```

Comment:
issue should be created to allow generalization beyond just npz here and elsewhere.

Reply:
Yeah and here we'd create sub-types of `Features` to encapsulate the various implementations